Player/Player.gd

```diff
@@ -2,26 +2,35 @@ extends CharacterBody2D
 
 signal interacted_with(interactable : Interactable)
 
-const SPEED = 50.0
+const ACCELERATION = 800
+const SPEED = 50
+const FRICTION = 300
+
 var interactables = []
 var closest_interactable
 
+@onready var animationPlayer = $AnimationPlayer
+@onready var animationTree = $AnimationTree
+@onready var animationState = animationTree.get("parameters/playback")
+
 func _physics_process(delta):
+    
     # Get the input direction and handle the movement/deceleration.
     # As good practice, you should replace UI actions with custom gameplay actions.
     var lr_direction = Input.get_axis("ui_left", "ui_right")
-
-    if lr_direction:
-        velocity.x = lr_direction * SPEED
-    else:
-        velocity.x = move_toward(velocity.x, 0, SPEED)
-
     var ud_direction = Input.get_axis("ui_up", "ui_down")
+    var input_vector = Vector2(lr_direction, ud_direction)
+    input_vector = input_vector.normalized()
```

Comment:
This condenses our velocity into a single output vector instead of modifying each component on the velocity individually (using `velocity.x` and `velocity.y`).